src/overlords/situational/dismantler.ts

```diff
@@ -0,0 +1,109 @@
+import {log} from '../../console/log';
+import {CombatSetups, Roles, Setups} from '../../creepSetups/setups';
+import {DirectiveModularDismantle} from '../../directives/targeting/modularDismantle';
+import {Pathing} from '../../movement/Pathing';
+import {OverlordPriority} from '../../priorities/priorities_overlords';
+import {profile} from '../../profiler/decorator';
+import {boostResources} from '../../resources/map_resources';
+import {Zerg} from '../../zerg/Zerg';
+import {Overlord} from '../Overlord';
+
+/**
+ * Spawns special-purpose dismantlers for transporting resources to/from a specified target
+ */
+@profile
+export class DismantleOverlord extends Overlord {
+
+	dismantlers: Zerg[];
+	directive: DirectiveModularDismantle;
+	target?: Structure;
+
+	requiredRCL: 4;
+
+	constructor(directive: DirectiveModularDismantle, target?: Structure, priority
+		= OverlordPriority.tasks.dismantle, boosted  = false) {
+		super(directive, 'dismantle', priority);
+		this.directive = directive;
+		// this.target = target || Game.getObjectById(this.directive.memory.targetId) || undefined;
+		this.dismantlers = this.zerg(Roles.dismantler, {
+			boostWishlist: boosted ? [boostResources.tough[3], boostResources.dismantle[3],
+				boostResources.move[3]] : undefined
+		});
+	}
+
+	init() {
+		// Spawn a number of dismantlers, up to a max
+		const MAX_DISMANTLERS = 2;
+		let setup;
+		if (!!this.directive.memory.attackInsteadOfDismantle) {
```

Comment:
i'm going to make an `addMemory()` method or something on the base overlord class that let you more easily wrap memory on a parent object so you don't have to keep cross-referencing directive memory, which is bad practice